src/Build/Evaluation/Expander.cs

```diff
@@ -775,47 +772,99 @@ private static int ScanForClosingQuote(char quoteChar, string expression, int in
         }
 
         /// <summary>
-        /// Add the argument in the StringBuilder to the arguments list, handling nulls
+        /// Add the argument represented by the slices to the arguments list, handling nulls
         /// appropriately.
         /// </summary>
-        private static void AddArgument(List<string> arguments, SpanBasedStringBuilder argumentBuilder)
+        private static void AddArgumentFromSlices(List<string> arguments, List<Tuple<int, int>> slices, string arg)
         {
-            // we reached the end of an argument, add the builder's final result
-            // to our arguments.
-            argumentBuilder.Trim();
-            string argValue = argumentBuilder.ToString();
+            // This shouldn't happen
```

Comment:
please elaborate on "why this shouldn't happen"

Reply:
There should always be at least one slice. If something goes here then something is wrong.
Maybe I should've been more explicit and throw instead or remove the check alltogether since I don't think it can happen.
It was mostly me covering my bases when starting to fiddle around with indices.